XMonad/Layout/IndependentScreens.hs

```diff
@@ -175,14 +247,39 @@ whenCurrentOn s pp = pp
 -- | If @vSort@ is a function that sorts 'WindowSpace's with virtual names, then @marshallSort s vSort@ is a function which sorts 'WindowSpace's with physical names in an analogous way -- but keeps only the spaces on screen @s@.
 marshallSort :: ScreenId -> ([WindowSpace] -> [WindowSpace]) -> ([WindowSpace] -> [WindowSpace])
 marshallSort s vSort = pScreens . vSort . vScreens where
-    onScreen ws = unmarshallS (tag ws) == s
+    onScreen ws = unmarshallS (W.tag ws) == s
     vScreens    = map unmarshallWindowSpace . filter onScreen
     pScreens    = map (marshallWindowSpace s)
 
+-- | Different from @marshallSort@, @marshallSort'@ does not convert 'WindowSpace'\'s physical names back to virtual names. 
```

Comment:
This comment seems wrong. Neither function is meant to change names, the only difference is what names the `vSort` functions works with.